go/chat/flipmanager.go

```diff
@@ -616,29 +616,47 @@ func (m *FlipManager) StartFlip(ctx context.Context, uid gregor1.UID, hostConvID
 }
 
 // MaybeInjectFlipMessage implements the types.CoinFlipManager interface
-func (m *FlipManager) MaybeInjectFlipMessage(ctx context.Context, msg chat1.MessageUnboxed,
-	convID chat1.ConversationID, topicType chat1.TopicType) {
+func (m *FlipManager) MaybeInjectFlipMessage(ctx context.Context, boxedMsg chat1.MessageBoxed,
+	inboxVers chat1.InboxVers, uid gregor1.UID, convID chat1.ConversationID, topicType chat1.TopicType) bool {
 	// earliest of outs if this isn't a dev convo, an error, or the outbox ID message
-	if topicType != chat1.TopicType_DEV || !msg.IsValid() || m.isHostMessageInfoMsgID(msg.GetMessageID()) {
-		return
+	if topicType != chat1.TopicType_DEV || boxedMsg.GetMessageType() != chat1.MessageType_FLIP ||
+		m.isHostMessageInfoMsgID(boxedMsg.GetMessageID()) {
+		return false
+	}
+	defer m.Trace(ctx, func() error { return nil }, "MaybeInjectFlipMessage: convID: %s", convID)()
+	// Update inbox for this guy
+	if err := m.G().InboxSource.UpdateInboxVersion(ctx, uid, inboxVers); err != nil {
+		m.Debug(ctx, "MaybeInjectFlipMessage: failed to update inbox version: %s", err)
+		// charge forward here, we will figure it out
+	}
+	// Unbox the message
+	conv, err := utils.GetUnverifiedConv(ctx, m.G(), uid, convID, types.InboxSourceDataSourceAll)
+	if err != nil {
+		m.Debug(ctx, "MaybeInjectFlipMessage: failed to get conversation for unbox: %s", err)
+		return true
+	}
+	msg, err := NewBoxer(m.G()).UnboxMessage(ctx, boxedMsg, conv.Conv, nil)
```

Comment:
won't these won't get exploded since we don't merge into storage?

Reply:
Why do we care then?